### ISSUE TITLE
Adding a domain.

### ISSUE THREAD

**Issue by Zaxter** [nsupdate-info/nsupdate.info#256]
I have a domain name that I have registered with a domain name registrar. I'm confused how to add this domain to my nsupdate.info `domains`. 

1. Do I need a hosting provider for this domain or just configuring the DNS in the domain name registrar would suffice? 

2. If a hosting is required, would it be possible to use cPanel based shared hosting?

**Comment by ThomasWaldmann** [comment]
you (or your dns provider) need to run a nameserver (e.g. bind) that can be dynamically updated (rfc2136) for your domain.

**Comment by Zaxter** [comment]
Thanks for replying @ThomasWaldmann . I'm still confused.

Now say I have a `mydomainname.com` with some domain registrar (like GoDaddy). They give me two name servers (eg. `ns1.something.com` & `ns2.something.com`). 

I figured to add this domain, I need to get the nameserver's IP and enter that in the domain form. After that I need to add the said configuration in my zone file and the nameserver's `named.conf` (or elsewhere).

Is this correct? If so, how can I get to my nameserver?

**Comment by ThomasWaldmann** [comment]
Offering dynamic updates to your domains is not quite a thing offered by all dns hosters, some offer just the bare minimum, but not advanced features as that. So you have to check for whether rfc2136 compatible updates are offered by whoever you choose.

If you host your own bind9 for your domain, you can implement that yourself. Our docs and the configuration examples for that should be quite helpful and do most of the job.

**Comment by dmsherazi** [comment]
@ThomasWaldmann  can you please provide ink for the configuration examples of bind9

**Comment by ThomasWaldmann** [comment]
if you add a new domain to nsupdate.info, it will show you bind9 configuration example on the result page.

**Comment by dmsherazi** [comment]
I am trying to install bind9 on digital ocean following this tutorial https://www.digitalocean.com/community/tutorials/how-to-configure-bind-as-a-private-network-dns-server-on-ubuntu-14-04

after this I would point Nameservers of my domain smartlink.com to the ip addresses of the bin9 nameservers IP

later I would add domain in nsupdate.info by putting the ip of the bind9 server in nameserver ip and smartlink.com as domain

is this the correct way?

is this correct proceedure?

**Comment by ThomasWaldmann** [comment]
you can maybe use some of that howto, but not all. it's not a private network dns, it should support v6 and dynamic updates. so use the sample config we show, if in doubt / if conflicts arise.

**Comment by dmsherazi** [comment]
can you give me alink or example how to setup bind 9


Best Regards,

              <http://www.dostmuhammad.com/>

Dost Muhammad Shah / Engr
info@dostmuhammad.com
engr.dost.mhd@gmail.com

+968 9777-5603
www.dostmuhammad.com

[image: Twitter] <http://twitter.com/dmsherazi> [image: Facebook]
<https://www.facebook.com/Dost.Muhammad.Shah> [image: Google +]
<https://plus.google.com/u/0/+DostMuhammadShah/posts> [image: LinkedIn]
<https://www.linkedin.com/in/dmsherazi> [image: Flickr]
<https://www.flickr.com/photos/54804393@N02/> [image: Github]
<https://github.com/dmsherazi> [image: Stack Overflow]
<http://stackoverflow.com/a/12335916/1155507> [image: Quora]
<https://www.quora.com/profile/Dost-Muhammad-Sherazi>

**Comment by ThomasWaldmann** [comment]
https://github.com/nsupdate-info/nsupdate.info/issues/256#issuecomment-211054621 that + google + official bind 9 documentation.

**Comment by dmsherazi** [comment]
How to setup the initial zone file?  which file would it be in bind9?

**Comment by ThomasWaldmann** [comment]
the initial zone file example is also shown after you add a domain on nsupdate.info.

**Comment by dmsherazi** [comment]
@ThomasWaldmann  I tried to setup the bind9 nameserver . When I am adding a domain and marking it as available I get this error on 

````
Apr 20 04:45:27 packer-ubuntu-15-10-amd64 named[13612]: client 104.236.203.198#60581/key smartlink.tk: updating zone 'smartlink.tk/IN': adding an RR at 'connectivity-test.smartlink.tk' A
Apr 20 04:45:27 packer-ubuntu-15-10-amd64 kernel: [10990.378382] audit: type=1400 audit(1461141927.656:9): apparmor="DENIED" operation="mknod" profile="/usr/sbin/named" name="/etc/bind/zones/smartlink.tk.jnl" pid=13615 comm="named" requested_mask="c" denied_mask="c" fsuid=108 ouid=108
Apr 20 04:45:27 packer-ubuntu-15-10-amd64 named[13612]: /etc/bind/zones/smartlink.tk.jnl: create: permission denied
Apr 20 04:45:27 packer-ubuntu-15-10-amd64 named[13612]: client 104.236.203.198#60581/key smartlink.tk: updating zone 'smartlink.tk/IN': error: journal open failed: unexpected error

````

**Comment by ThomasWaldmann** [comment]
"Permission denied" plus a filename should be pretty clear.

**Comment by finite9** [comment]
Im in the same position (I think) and I read this thread but still dont get it.  Can you explain in layman terms?

Ive got ddns working fine with an nsupdate.info address.  I just bought my own .net domain from Namecheap.  They offer ddns as well, but I want to continue using nsupdate.  Namecheap also supports ddclient and they have a hash password thing.  Forgetting the details for a minute, can I do the following: Update my Namecheap domain with my current IP address from nsupdate's service by entering my domain details on nsupdate.info?  Im not even sure that what I want to do is possible?

Secondly, if this _is_ possible... do I just enter my domain name and the ip of Namecheaps DNS for the above to work, assuming that they do support rfc2136?

I don't have my own bind9 server and don't want one.

**Comment by ThomasWaldmann** [comment]
You should know / find out if they support rfc2136 or not (if unsure, assume they do not).

If they don't support rfc2136 (dynamic updates, this is something different than dyndns updates), nsupdate.info can not update them (as it does for e.g. the nsupdate.info DNS server).

**Comment by dmsherazi** [comment]
@finite9  I had a namecheap domain and I tried the same , but it failed. Finally I had to host my own bin9 server which I did on the same server where I installed nsupdate.info.

**Comment by ThomasWaldmann** [comment]
Closing this, IIRC the docs point out the requirements clear enough and also we show ready-to-use configs for bind9.
